lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -363,3 +373,47 @@ def _assert_retry_not_called(self, mock_retry):
         Verifies the task was not retried.
         """
         self.assertFalse(mock_retry.called)
+
+
+@ddt.ddt
+class ComputeGradesForCourseTest(HasCourseWithProblemsMixin, ModuleStoreTestCase):
+    """
+    Test compute_grades_for_course task.
+    """
+
+    ENABLED_SIGNALS = ['course_published', 'pre_publish']
+
+    def setUp(self):
+        super(ComputeGradesForCourseTest, self).setUp()
+        self.users = [UserFactory.create() for _ in xrange(12)]
+        self.set_up_course()
+        for user in self.users:
+            CourseEnrollment.enroll(user, self.course.id)
+
+    @ddt.data(*xrange(0, 12, 3))
+    def test_behavior(self, batch_size):
+        result = compute_grades_for_course.delay(
+            course_key=six.text_type(self.course.id),
+            batch_size=batch_size,
+            offset=4
+        )
+        self.assertTrue(result.successful)
+        self.assertEqual(
+            PersistentCourseGrade.objects.filter(course_id=self.course.id).count(),
+            min(batch_size, 8)  # No more than 8 due to offset
+        )
+        self.assertEqual(
+            PersistentSubsectionGrade.objects.filter(course_id=self.course.id).count(),
+            min(batch_size, 8)  # No more than 8 due to offset
+        )
+
+    @ddt.data(*xrange(1, 12, 3))
+    def test_database_calls(self, batch_size):
+        per_user_queries = 16 * min(batch_size, 6)  # No more than 6 due to offset
```

Comment:
"per_user_queries" is not referenced. I think you meant to reference it on line 413, so I'll make that change in #14731 because I need to up the number of query calls there.